.gitignore

```diff
@@ -1,6 +1,10 @@
 # Built binaries.
 oasis-indexer
 
+# Autogenerated Go code.
+api/v1/types
```

Comment:
Hmm on second thought, mind leaving the types/http server checked in? It makes it easier to navigate the repo, and hopefully it helps people realize earlier if their types are generated incorrectly :)

Reply:
> would simply having **/*.gen.go suffice?

Yes, after I rebased on top of your changes, that's exactly how this line changed :+1: 

> mind leaving the types/http server checked in? It makes it easier to navigate the repo, and hopefully it helps people realize earlier if their types are generated incorrectly :)

I've waffled on this and ultimately would like to say no, but I feel a little bad about it, so here's an over-extended explanation (and hopefully some improvements for your workflow).

To your second point, I now added a check to the Makefile rule to make sure we're all using the same version.
To your first point ... I do like that argument for GitHub browsing. For more serious, local browsing though, surely it doesn't matter much because you always have the autoegenerated files present (but ignored by git)?
Another thing I like about having files checked in is that one can see in github reviews what the impact of .yaml changes on .go is.

That said, I'd quite like to keep these files ignored. I had them checked in initially, and it really clashed with the git workflow: Whenever you do anything with the .yaml file (commit, revert, god forbid rebase), you have to do it three times. Interactive rebases (commit reordering) became more painful. And most annoying of all, if you make a change in the yaml and commit it, but forget to commit the other two files, then later make more changes to .yaml and now commit all three files ... now the first change is spread across two commits: the .yaml change is in one commit, and the .go change is in the second commit, commingled with the second change. That can be untangled but it's a noticeable amount of annoying git surgery.   
This setup is also in line with BigCo best practices, where they check in the .proto or .thrift or whatever languaeg-agnostic type specs, and not their autogenerated artifacts. Not all BigCo wisdom generalizes to small companies, but I don't see this one as size-specific.

Last thing: To make local browsing seamless, you can add a git hook so that your local files are up-to-date after every checkout: Create a file oasis-indexer/.git/post-checkout, make it executable, and put this in it:
```bash
#!/bin/bash

# Autogenerate Go code from OpenAPI spec
make --dry-run --question codegen-go 2>/dev/null
if (( $? < 2 )); then  # make exits with 2 if rule does not exist
  make codegen-go
fi
```